docs/DafnyRef/Attributes.md

```diff
@@ -344,136 +302,206 @@ recursion was explicitly requested.
+The first problem considers all assertions that are on the bloc containing the `assert {:focus} X;`.
+The second problem considers all remaining assertions[^second-focus-assertion-batch].
+
+[^second-focus-assertion-batch]: More precisely, any assertion on an execution path that does not go through the assertion with `{:focus}` will be counted in the second problem, all the others will be counted in the first problem.
+
+For example:
+```
+method doFocus(x: bool) returns (y: int) {
+  y := 1;
+  assert y == 1;    // Assumption in batch #1, Assertion in batch #2
+  if x {
+    assert {:focus} true;
+    y := 2;
+    assert y == 2; // Assertion in batch #1, Assumption in batch #2
+  } else {
+    assert y == 1; // Assumption in batch #1, Assertion in batch #2
+  }
+  assert y >= 1;   // Assumption in batch #1, Assertion in batch #2
+}
+```
+
+### 22.3.2. `{:split_here}` {#sec-split_here}
+`assert {:split_here} X;` splits verification into two [assertion batches](#sec-assertion-batches).
+It verifies the code leading to this point (excluded) in a first assertion batch,
+and the code leading from this point (included) to the next `{:split_here}` or until the end in a second assertion batch.
+It might help with timeouts.
+It might also occasionally double-report errors.
+
+```
+method doSplitHere(x: bool) returns (y: int) {
+  y := 1;
+  assert y == 1;   // Assertion in batch #1, Assumption in batch #2
+  assert {:split_here} true;
+  assert y == 1;   // Assumption in batch #1, Assertion in batch #2
+  assert y >= 1;   // Assumption in batch #1, Assertion in batch #2
+}
+```
+
+### 22.3.3. `{:subsumption n}`
+Overrides the `/subsumption` command-line setting for this assertion.
+
+
+<!--
+This is not working in Dafny and shouldn't be documented until so.
+### 22.3.4. `{:msg <string>}`
+     
+Prints <string> rather than the standard message for assertion failure.
+Also applicable to requires and ensures declarations.
+
+```
+method doFocus(x: bool) returns (y: int) {
+  y := 1;
+  assert {:msg "y is indivisible"} y != 1;    //
+}
+```
+-->
+
+
+
+## 22.4. Attributes on variable declarations
+
+### 22.4.1. `{:assumption}`
+This attribute can only be placed on a local ghost bool
+variable of a method. Its declaration cannot have a rhs, but it is
+allowed to participate as the lhs of exactly one assignment of the
+form: `b := b && expr;`. Such a variable declaration translates in the
+Boogie output to a declaration followed by an `assume b` command.
+See [@LeinoWuestholz2015], Section 3, for example uses of the `{:assumption}`
+attribute in Boogie.
+
+### 22.4.2. `{:existential}`
+Marks a global Boolean variable as existentially quantified. If
+used in combination with option `/contractInfer`, Boogie will check
+whether there exists a Boolean assignment to the existentials
+that makes all verification conditions valid.  Without option
+`/contractInfer` the attribute is ignored.
```

Comment:
Does this option work with Dafny?

Reply:
I have no idea, as it's the only documentation that exists about it. It should work however, since attributes are passed onto Boogie. I suggest we remove deprecate this attribute though?